xamarin/Xamarin.Forms
# [Android,iOS] Fix setting position on Ctor and OnAppearing on CarouselView

### Description of Change ###

Fix issues when setting the Position on CarouselView on the ctor, on a different order or on appearing when the itemsource is set later.

Fix a small issue with `IndicatorView` when setting collection after the position would not update to the current position.

### Issues Resolved ### 


- fixes #10170 
- fixes #10291 

### API Changes ###

 None

### Platforms Affected ### 

- iOS
- Android

### Behavioral/Visual Changes ###

Should scroll to the item position on page load by setting on ctor or onappearing

### Before/After Screenshots ### 


Not applicable

### Testing Procedure ###

Go to gallery CarouselItemsGallery

pick some of the options 
"CarouselView SetPosition Appearing",
"CarouselView SetPosition Ctor" 
"CarouselView SetPosition Ctor No Animation"

Should always go to the Item4 (3rd index)

### PR Checklist ###
